docs/csharp/programming-guide/classes-and-structs/extension-methods.md

```diff
@@ -59,10 +60,38 @@ using System.Linq;
   
  [!code-csharp[csProgGuideExtensionMethods#5](~/samples/snippets/csharp/VS_Snippets_VBCSharp/csProgGuideExtensionMethods/cs/extensionmethods.cs#5)]  
   
+## Common Usage Patterns
+
+### Collection Functionality
+  In the past it was common to create "Collection Classes" that implemented the <xref:System.Collections.Generic.IEnumerable%601?displayProperty=nameWithType> interface for a given type and contained functionality that acted on collections of that type. While there is nothing wrong with creating this type of collection object, the same functionality can be achieved by using an extension on the <xref:System.Collections.Generic.IEnumerable%601?displayProperty=nameWithType>. Extensions have the advantage of allowing the functionality to be called from any collection such as an <xref:System.Array?displayProperty=nameWithType> or <xref:System.Collections.Generic.List%601?displayProperty=nameWithType> that implements <xref:System.Collections.Generic.IEnumerable%601?displayProperty=nameWithType> on that type. An example of this using an Array of Int32 can be found [earlier in this article](#orderby-example).
+
+### Layer-Specific Functionality
+  When using an Onion Architecture or other layered application design it is common to have a set of Domain Entities or Data Transfer Objects that can be used to communicate across application boundaries. These objects generally contain no functionality, or only minimal functionality that applies to all layers of the application. Extension methods can be used to add functionality that is specific to each application layer without loading the object down with methods not needed or wanted in other layers.
+
+```aspx-csharp
+public class DomainEntity
+{
+    public Int32 Id { get; set; }
+    public String FirstName { get; set; }
+    public String LastName { get; set; }
+}
+
+static class DomainEntityExtensions
+{
+    static String FullName(this DomainEntity value)
```

Comment:
I have the same comment on all these samples. Our coding standard is to use the built in C# types rather than the library types:

```suggestion
    public int Id { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
}

static class DomainEntityExtensions
{
    static string FullName(this DomainEntity value)
```

Reply:
Thanks for making these updates @bsstahl